.github/workflows/build_and_test.yml

```diff
@@ -194,6 +193,55 @@ jobs:
         if: matrix.build_type == 'debug'
         uses: ./.github/actions/save-coverage-data
 
+  check-codestyle-rust:
+    strategy:
+      fail-fast: false
+      matrix:
+        # If we want to duplicate this job for different
+        # Rust toolchains (e.g. nightly or 1.37.0), add them here.
+        rust_toolchain: [1.58]
```

Comment:
This variable is not used anywhere except the UI, so I've got confused by it a bit, considering that we're using 1.58 rustc for running `cargo clippy`.

Instead, we use latest stable rustc, 1.62, [preinstalled](https://github.com/neondatabase/neon/runs/7317743544?check_suite_focus=true#step:1:10) on `ubuntu-latest` by GH team.
Newer clippy has more lints in it and fails currently in CI.

We should either take the `Install rust toolchain` step from `codestyle.yml` or reuse our `neondatabase/rust:1.58` image for this job. 
I don't think using latest (volatile) stable rustc for clippy linting is worth it, since we bump the toolchain manually over the time.

Reply:
I removed the runs-on directive, so clippy now runs in the same environment as the other jobs in the workflow.

That means that we no longer run clippy on macOS. I think that's OK. It was a bit weird to check that clippy runs on macOS, but not run any of the other tests. If we want to support macOS, and run it in the CI, I think we should try to run all the tests.